code/datums/components/cult_ritual_item.dm

```diff
@@ -0,0 +1,463 @@
+
+	if(cultist.incapacitated())
+		return FALSE
+
+	if(cultist.stat == DEAD)
+		return FALSE
+
+	return TRUE
+
+/*
+ * Helper to check if a rune can be scribed by a cultist.
+ * Used in between inputs of [do_scribe_rune] for sanity checking.
+ *
+ * tool - the parent - the item being used to scribe the rune, casted to item
+ * cultist - the mob making the rune
+ */
+/datum/component/cult_ritual_item/proc/can_scribe_rune(obj/item/tool, mob/living/cultist)
+	if(!IS_CULTIST(cultist))
+		to_chat(cultist, span_warning("[tool] is covered in unintelligible shapes and markings."))
+		return FALSE
+
+	if(QDELETED(tool) || !cultist.is_holding(tool))
+		return FALSE
+
+	if(cultist.incapacitated() || cultist.stat == DEAD)
+		to_chat(cultist, span_warning("You can't draw a rune right now."))
+		return FALSE
+
+	if(!check_rune_turf(get_turf(cultist), cultist))
+		return FALSE
+
+	return TRUE
+
+/*
+ * Checks if a turf is valid for having a rune placed there.
+ *
+ * target - the turf being checked
+ * cultist - the mob placing the rune
+ */
+/datum/component/cult_ritual_item/proc/check_rune_turf(turf/target, mob/living/cultist)
+	if(isspaceturf(target))
+		to_chat(cultist, span_warning("You cannot scribe runes in space!"))
+		return FALSE
+
+	if(locate(/obj/effect/rune) in target)
+		to_chat(cultist, span_cult("There is already a rune here."))
+		return FALSE
+
+	var/area/our_area = get_area(target)
+	if((!is_station_level(target.z) && !is_mining_level(target.z)) || (our_area && !(our_area.area_flags & CULT_PERMITTED)))
+		to_chat(cultist, span_warning("The veil is not weak enough here."))
+		return FALSE
+
+	return TRUE
+
+/*
+ * Helper to check a cultist is located in one of the ritual / summoning sites.
+ *
+ * cultist - the mob making the rune
+ * cult_team - the team of the mob making the rune
+ * fail_if_last_site - whether the check fails if it's the last site in the summoning list.
+ */
+/datum/component/cult_ritual_item/proc/check_if_in_ritual_site(mob/living/cultist, datum/team/cult/cult_team, fail_if_last_site = FALSE)
+	var/datum/objective/eldergod/summon_objective = locate() in cult_team.objectives
+	var/area/our_area = get_area(cultist)
+	if(!summon_objective)
+		to_chat(cultist, span_warning("There are no ritual sites on this station to scribe this rune!"))
+		return FALSE
+
+	if(!(our_area in summon_objective.summon_spots))
+		to_chat(cultist, span_warning("This veil is not weak enough here - it can only be scribed in [english_list(summon_objective.summon_spots)]!"))
+		return FALSE
+
+	if(fail_if_last_site && summon_objective.summon_spots.len <= 1)
+		to_chat(cultist, span_warning("This rune cannot be scribed here - the ritual site must be reserved for the final summoning!"))
+		return FALSE
+
+	return TRUE
+
+/*
+ * Removes all shields from the shields list.
+ */
+/datum/component/cult_ritual_item/proc/cleanup_shields()
+	for(var/obj/structure/emergency_shield/cult/narsie/shield as anything in shields)
+		LAZYREMOVE(shields, shield)
```

Comment:
```suggestion
		LAZYREMOVE(shields, shield)
```
instead, just do LAZYCLEARLIST(shields) at the end of the proc, it's more optimal
Additionally you can do QDEL_LAZYLIST(shields)

Reply:
It's not a big deal at all though so I'll just be merging this anyways